proof-defns/proof_defns.tex

```diff
@@ -0,0 +1,438 @@
+%\end{definition}
+
+Note: symmetric distance is the metric which is used in the OpenDP library, while the definition symmetric difference is only included for completeness in this document.
+
+% still working on this paragraph
+Because there is a bijection between histograms and multisets, we can also define the \emph{symmetric distance} between $u$ and $v$ as the $\ell_1$ distance between the histograms for $u$ and $v$, denoted $h_u$ and $h_v$ (see Definition \ref{defn:histogram}). Then, we equivalently obtain that the symmetric distance between $u$ and $v$ is
+$$d_{\text{Sym}}(u,v) = \lVert h_{v} - h_{w}\rVert_1 = \sum_{z\in \mathcal{X}} |h_u(z) - h_{v}(z)|.$$
+
+The second equality follows from the definition of $\ell_1$ distance.
+
+\silvia{Add the above as claim?}
+
+\connor{Since these definitions are all written by us, I think we could say that we're defining symmetric distance as both things.}
+
+\medskip
+\textbf{Alternative definition of symmetric distance.}
+
+Let $u,v$ be vectors of elements drawn from domain $\mathcal{X}$. Define $m_v(\ell)$ as the multiplicity of element $\ell$ in vector $v$. For example, if $v$ contains five instances of the number ``21'', then $m_v(21) = 5$.
+
+An alternative definition of symmetric distance, then, is $$d_{\text{Sym}}(u,v) = \sum_{z\in \mathcal{X}} |m_u(z) - m_v(z)|.$$
+
+\questionc{Which definition should we go with? Original definition, alternative definition, or both?}
+
+\begin{claim}
+Symmetric distance is a metric.
+\end{claim}
+
+Note that null data values are still counted in the symmetric distance. Adding or removing null values still influences the count.
+
+\subsection{The path property of symmetric distance on sized domains}
+
+\questionc{Given that we are avoiding texttt notation in this document, should we change \texttt{SizedDomain} for a mathematical definition of sized domain in the three lemmas that follow?}
+
+\begin{lemma}[Path property of $d_{Sym}$ on \texttt{SizedDomain}]
+    For any two vectors $v, w$ of the same length (that is, for any two vectors $v, w \in$ \texttt{SizedDomain(D)} for some domain \texttt{D}), $d_{Sym}(v,w)$ is an even integer; i.e., $d_{Sym}(v,w) = 2k$ for some integer $k \geq 0$. Moreover, there exist $k$ vectors $v=v_0, v_1, \ldots, v_k=w$ such that $d_{Sym}(v_i,v_{i+1})=2$ for all $i$.
+\end{lemma}
+
+\begin{lemma}
+    For any two vectors $v, w \in$ \texttt{SizedDomain(D)}, $d_{Sym} = 2$ if and only if we can change one element of $v$ to obtain $v'$ such that $\MultiSet(v') = \MultiSet(w)$.
+\end{lemma}
+
+The above lemma follows directly from the definition of symmetric distance. 
+
+\begin{lemma}
+Given a function $f$ on input domain \texttt{SizedDomain(D)}, if $\dout(f(v), f(w)) \leq c$ for all vectors $v, w$ such that $d_{Sym}(v, w) = 2$, then $f$ is $c/2$-stable.
```

Comment:
Sorry, what do you mean by the first sentence?

Reply:
You are right, it should be d_Y and not d_out. So v and w are coming from SizedDomain(D, n), which is why we thought it was redundant to add "of the same length", but re-phrased it a bit now. We set if to 2 because it is paired up with the two previous lemmas, which allow to make "small" jumps of 2 to reach any even distance.